src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -20,12 +23,59 @@ export default class GroupComparisonStore {
             for (const sample of this.resultsViewPageStore.samples.result!) {
                 sampleIdentifierToSample.set(sample, sample.sampleId, sample.studyId);
             }
+            let colors = _.cloneDeep(COLORS);
             return Promise.resolve(
-                this.resultsViewPageStore.sampleGroups.result!.map(group=>({
+                this.resultsViewPageStore.sampleGroups.result!.map((group)=>({
                     name: group.name,
-                    samples: group.sampleIdentifiers.map((si:SampleIdentifier)=>sampleIdentifierToSample.get(si.sampleId, si.studyId))
+                    samples: _.reduce(group.sampleIdentifiers,(samples:Sample[], si:SampleIdentifier)=>{
+                        if(sampleIdentifierToSample.get(si.sampleId, si.studyId)){
+                            samples.push(sampleIdentifierToSample.get(si.sampleId, si.studyId)!);
+                        }
+                        return samples;
+                    },[]),
+                    color: group.color ? group.color : colors.shift()!,
```

Comment:
on second thought, if you had a ton of sample groups then you could run out of colors right? so maybe
```
let colorIndex = 0;
return Promise.resolve(
...
color: group.color ? group.color : COLORS[(colorIndex++) % COLORS.length]
...
)
```